src/ObjCRuntime/INativeObject.cs

```diff
@@ -20,6 +20,15 @@ static public IntPtr GetHandle (this INativeObject self)
 		{
 			return self == null ? IntPtr.Zero : self.Handle;
 		}
+
+		static public IntPtr GetNonNullHandle (this INativeObject self, string argumentName)
+		{
+			if (self == null)
+				ThrowHelper.ThrowArgumentNullException (argumentName);
+			if (self.Handle == IntPtr.Zero)
+				ThrowHelper.ThrowObjectDisposedException (self);
```

Comment:
We have an argument name - I know it's an existing API, but shouldn't we put that into the exception to make chasing down an error easier?